### PR TITLE
Add support for a list of paths in each PluginType

### PR DESCRIPTION
**Pull request by Gee19**
Refs #SDE-3585

### CODE REVIEW

**Comment by aeldamaty** [comment]
+1 looks good to me

File: yantra/manager.py [hunk 1]
```diff
@@ -45,21 +45,22 @@ def _get_modules(self):
         """Discover all modules in the path"""
         plugin_modules = []
 
-        # walk the plugins folder
-        for root, subdir, files in os.walk(self.plugin_type.path):
+        for p in self.plugin_type.paths:
```
Comment (hanifvirani) [inline yantra/manager.py]: Rename `p` to `path`

File: yantra/manager.py [hunk 2]
```diff
@@ -45,21 +45,22 @@ def _get_modules(self):
...
+            # walk the plugins folder
+            for root, subdir, files in os.walk(p):
```
Comment (hanifvirani) [inline yantra/manager.py]: This method is getting a bit ridiculous at this point. Might be good to extract the walking part into another method that takes a plugin.

**Comment by Gee19** [comment]
I removed the existing _get_modules method because I felt it was an unnecessary abstraction after I extracted the folder walking to it's own method(populate_modules). I am now calling populate_modules for each path inside the get_plugins method. 

Also should I return modules from the populate_modules method or leave it the way it is? (Mutating the object passed in)

Let me know what you think @hanifvirani @aeldamaty

**Comment by aeldamaty** [comment]
+1. If @hanifvirani has any preferences for way the method should act that's fine. Both sound good to me.

**Comment by hanifvirani** [comment]
Yeah, I would leave the `get_plugins` method the way it was and just refactor `_get_modules` into two pieces.

**Comment by Gee19** [comment]
@hanifvirani Done. I'm still unhappy with populate_plugin_modules_from_path mutating the list being passed to it rather than just returning a module and _get_modules appending to plugin_modules.

**Comment by hanifvirani** [comment]
As discussed in person, turns out we don't need this.

**Pull request closed without merging**